### PR TITLE
Milestone rank

### PR DESCRIPTION
**Pull request by g4s8**
#575 - Added comparator to give higher rank to milestoned issues

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1521](https://github.com/zerocracy/farm/issues/1521) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1521](https://github.com/zerocracy/farm/issues/1521) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2131204008); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/zerocracy/pm/staff/ranks/RnkGithubMilestoneTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class RnkGithubMilestoneTest {
+    @Test
+    public void sortMilestonedIssuesFirst() throws Exception {
+        final Github github = new MkGithub().relogin("test");
+        final Repo repo = github.repos().create(
+            new Repos.RepoCreate("milestones", false)
+        );
+        final List<String> jobs = new ArrayList<>(3);
```
Comment (krzyk) [inline src/test/java/com/zerocracy/pm/staff/ranks/RnkGithubMilestoneTest.java]: @g4s8 I think it is better to use `LinkedList`, no need to initialize size (and doing initialization in unit tests is pointless)

File: src/main/java/com/zerocracy/pm/staff/ranks/RnkGithubMilestone.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import java.util.Comparator;
+import org.cactoos.BiFunc;
+import org.cactoos.func.StickyBiFunc;
+import org.cactoos.func.UncheckedBiFunc;
+
+/**
+ * Give higher rank to Github issues with milestone with
+ * earlier {@code due_date} field.
+ *
+ * @since 1.0
```
Comment (krzyk) [inline src/main/java/com/zerocracy/pm/staff/ranks/RnkGithubMilestone.java]: @g4s8 should we use major versions only, or minor also? (like `0.28`)

Comment (g4s8) [reply src/main/java/com/zerocracy/pm/staff/ranks/RnkGithubMilestone.java]: @krzyk yes, we've swtiched to major-only in #1420

**Comment by g4s8** [comment]
@krzyk fixed in https://github.com/zerocracy/farm/pull/1521/commits/1c648feaef9a513218cd7017d462e8ee7780d32e, please look again

**Comment by krzyk** [comment]
@g4s8 thanks

**Comment by krzyk** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/zerocracy/farm/issues/1521#issuecomment-410441194))

@g4s8 The job [#1521](https://github.com/zerocracy/farm/issues/1521) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@krzyk According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews

**Comment by krzyk** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@krzyk thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**